automagicallyexecute/index.ts

```diff
@@ -28,6 +29,16 @@ const run = async (): Promise<void> => {
       return
     }
 
+    const sha = getVariable('Build.SourceVersion')
+    if (!sha) {
+      setResult(TaskResult.Failed, 'sha is required')
+    }
+
+    const ref = getVariable('Build.SourceBranch')
+    if (!ref) {
+      setResult(TaskResult.Failed, 'ref is required')
+    }
```

Comment:
not sure if we shouldn't keep them optional as well to not get rid of one requirement to introduce another one